internals/api/credential.go

```diff
@@ -62,27 +66,28 @@ type CredentialType string
 
 // Credential types
 const (
-	CredentialTypeKey        CredentialType = "key"
-	CredentialTypeAWS        CredentialType = "aws"
-	CredentialTypeBackupCode CredentialType = "backup-code"
+	CredentialTypeKey               CredentialType = "key"
+	CredentialTypeAWS               CredentialType = "aws"
+	CredentialTypeBackupCode        CredentialType = "backup-code"
+	CredentialTypeGCPServiceAccount CredentialType = "gcp-service-account"
```

Comment:
Because on GCP this method is _only_ usable for service accounts. For user accounts it does not work and probably will never. For AWS we use the more general concept of roles, which is more generally usable.

Maybe a more suitable name for `aws` would be something like `aws-role` because it would be possible to create a credential for an AWS IAM user.